include/glow/Backend/BackendUtils.h

```diff
@@ -120,5 +120,27 @@ class RuntimeBundle {
 };
 } // namespace runtime
 
+/// If \p PH is an output placeholder, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool isOutput(const Placeholder *PH, const Function &F);
+
+/// If \p PH is an input placeholder, \returns true.
+bool isInput(const Placeholder *PH, const Function &F);
+
+/// If \p PH is an output placeholder in the function \p F, \returns true.
+/// This is determined by checking if the PH has a user which uses the PH as an
+/// overwritten input.
+bool isOutput(const Placeholder *PH, const IRFunction &F);
+
+/// If \p PH is an input placeholder in the function \p F, \returns true.
+bool isInput(const Placeholder *PH, const IRFunction &F);
+
+/// Convert placeholders to be ordered as input|inputOutput|output|neither.
+/// Packed into {Placeholder *, isInput, isOutput}.
+template <typename FUN, typename ARR>
```

Comment:
Please add comments about `FUN` and `APR` parameters

Reply:
BTW, how does it work for a call from `AllocationsInfo.cpp`? This header does not provide the implementation of this template function. How does `AllocationsInfo.cpp` manage to call it and find a proper implementation?

Reply:
> BTW, how does it work for a call from `AllocationsInfo.cpp`? This header does not provide the implementation of this template function. How does `AllocationsInfo.cpp` manage to call it and find a proper implementation?

I tried to comment the declaration in the BackendUtils.h its a compile error and If I comment the implementation in the BackendUtils.cpp its a linking error. I searched 'BackendUtils.cpp' and it is not included in anywhere else. Still trying to figure out why it works.